integration-libs/punchout/core/services/punchout-auth.service.ts

```diff
@@ -47,19 +57,36 @@ export class PunchoutAuthService {
     );
   }
 
-  loginWithToken(accessToken: string, userId: string): void {
+  loginWithToken(accessToken: string): void {
     // Code mostly based on auth lib we use and the way it handles token properties
     this.authStorageService.setItem(ACCESS_TOKEN, accessToken);
     this.authStorageService.setItem(
       ACCESS_TOKEN_STORED_AT,
       Date.now().toString()
     );
-    this.userIdService.setUserId(userId);
+    this.authStorageService.setItem(TOKEN_TYPE, 'Bearer');
+    this.userIdService.setUserId(OCC_USER_ID_CURRENT);
```

Comment:
setting 'current' fixed all the 'token revoking failure': CXSPA-9871 Token not revoked when silently logout user.
It sems to be the standard practice to set current rather than user id string. Not sure about the reason.